app/code/core/Mage/CatalogInventory/Model/Stock/Item.php

```diff
@@ -524,9 +524,10 @@ public function suggestQty($qty)
      * @param mixed $qty quantity of this item (item qty x parent item qty)
      * @param mixed $summaryQty quantity of this product
      * @param mixed $origQty original qty of item (not multiplied on parent item qty)
+     * @param mixed $quoteItem quote item
      * @return Varien_Object
      */
-    public function checkQuoteItemQty($qty, $summaryQty, $origQty = 0)
+    public function checkQuoteItemQty($qty, $summaryQty, $origQty = 0, $quoteItem = null)
```

Comment:
Since this PR is modifying a public method, it will break the signature of every method the inherits it. Because of this we have to target `next` branch.